packages/mephisto-task/src/live.js

```diff
@@ -24,27 +24,67 @@ const AGENT_STATUS = {
   MEPHISTO_DISCONNECT: "mephisto disconnect",
 };
 
+const STATUS_TO_TEXT_MAP = {
+  [AGENT_STATUS.EXPIRED]:
+    "This task is no longer available to be completed. " +
+    " Please return it and try a different task",
+  [AGENT_STATUS.TIMEOUT]:
+    "You took to long to respond to this task, and have timed out. " +
+    "The task is no longer available, please return it.",
+  [AGENT_STATUS.DISCONNECT]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.RETURNED]:
+    "You have disconnected from our server during the duration of the task. " +
+    "If you have done substantial work, please reach out to see if we can recover it. ",
+  [AGENT_STATUS.PARTNER_DISCONNECT]:
+    "One of your partners has disconnected while working on this task. We won't penalize " +
+    "you for them leaving, so please submit this task as is.",
+  [AGENT_STATUS.MEPHISTO_DISCONNECT]:
+    "Our server appears to have gone down during the " +
+    "duration of this Task. Please send us a message if you've done " +
+    "substantial work and we can find out if the task is complete enough to " +
+    "compensate.",
+};
+
 const useMephistoLiveTask = function (props) {
   const [connectionStatus, setConnectionStatus] = React.useState("");
-  const [agentState, setAgentState] = React.useState(null);
   const [agentStatus, setAgentStatus] = React.useState(null);
 
   const defaultConnectionHook = useMephistoSocket;
   const useConnectionHook = props.customConnectionHook || defaultConnectionHook;
 
   const taskProps = useMephistoTask();
+
+  const { initialTaskData } = taskProps;
+
+  function onMessageReceived(message) {
+    const { state, message_id, timestamp, ...remaining_message } = message;
+    if (state !== undefined) {
+      props.onStateUpdate && props.onStateUpdate({ state });
+    }
+    if (Object.keys(remaining_message).length != 0) {
+      props.onMessageReceived && props.onMessageReceived(remaining_message);
```

Comment:
- I suggest we rename the local `onMessageReceived` function here to `handleMessageReceived`. It's serving more as [a handler than a callback](https://stackoverflow.com/a/2069800) in that its functionality is not passed around the code. The `on*` prefix initially confused me as it led me to think otherwise.
  - This also more clearly differentiates it from `props.onMessageReceived` - which is indeed a callback given that it's passed via the `props` argument.

- Is there any reason we don't want to pass down `message_id` and `timestamp` down to the callback? Could there be a use case where the end user requires them?
- Given the gated clause of `Object.keys(remaining_message).length != 0`, we'll never invoke onMessageReceived when there's only a state update. Just verifying that this is intended behavior?

- I want to challenge the object model here a bit. Correct me if I'm wrong: We have `raw_messages` (e.g. `initialTaskData.raw_messages`), which produce either `state`s or `message`s or both. A `message` is defined simply as any `raw_message` that has a field other than `state, message_id, timestamp`. A state update is simply any message that includes a `state` field. Correct?
  - What's also a bit confusing here is that [in the backend code](https://github.com/facebookresearch/Mephisto/blob/6e2a02d94ee29dac7d5ba501bd3c5b17eea4a973/mephisto/abstractions/blueprints/parlai_chat/parlai_chat_agent_state.py#L61), the `raw_messages` being sent over are called retrieved from a variable called `messages`. I think we could maybe use this refactoring as an opportunity to be more consistent, precise and defined about the ontology of messages.
  - Note: I think `state` here should be `status` (see earlier comment in this file), though I'm using `state` in this comment to keep consistent with the code in reference, even if a typo
  - Also are all `packets` a `raw_message` or is there some distinction between those two? My guess is no, because packets also include things such as heartbeats that secure the connection.

Reply:
> rename the local onMessageReceived function

Surely!

> Is there any reason we don't want to pass down message_id and timestamp down to the callback?

These are added by Mephisto, so I stripped them back out. I can add them back in though, as I imagine they're useful to some.

> we'll never invoke onMessageReceived when there's only a state update

Correct - this is if we choose to split messages and state updates on the `mephisto-task` level. Can discuss more on this though.

> object model for `raw_messages`

Yeah this is a fair discussion and worth refactoring. `raw_messages` in this case cover all of what we consider to be "live data" which includes messages that are 'state' updates (Both `LIVE_DATA` packet types). All other packets aren't included in this. I think it makes sense for me to rename all occurrences across everywhere to refer to these received 'messages' as receiving 'live_data', and then 'raw_messages' can become 'past_live_data' instead.